spec/providers/database.spec.ts

```diff
@@ -137,6 +137,12 @@ describe('DeltaSnapshot', () => {
       populate({ myKey: 'foo', myOtherKey: 'bar' }, { myKey: null });
       expect(subject.val()).to.deep.equal({ myOtherKey: 'bar' });
     });
+
+    // Regression test: .val() was returning array of nulls when there's a property called length
```

Comment:
Can you add "BUG(37683995)" in this comment, for our future reference if we want to remember more details?